### PR TITLE
test: [IV] Backend API Functional tests

### LINKED ISSUE #913: Functional tests for Integrated Vectorization

### PR DESCRIPTION
**Pull request by komalg1**
## Purpose

* This PR adds functional tests for Integrated Vectorization backend api
* Resolves #913 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

Test the code

* Run unit tests
* Run functional tests
* Deploy to Azure

## What to Check
Verify that the following are valid
* All tests passing
* Admin + webapp working

### CODE REVIEW

File: code/tests/functional/tests/backend_api/integrated_vectorization_custom_conversation/test_iv_question_answer_tool.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+
+    # then
+    verify_request_made(
+        mock_httpserver=httpserver,
+        request_matcher=RequestMatcher(
+            path=f"/indexes('{app_config.get('AZURE_SEARCH_INDEX')}')/docs/search.post.search",
+            method="POST",
+            json={
+                "search": "What is the meaning of life?",
+                "top": 5,
```
Comment (cecheta) [inline code/tests/functional/tests/backend_api/integrated_vectorization_custom_conversation/test_iv_question_answer_tool.py:175]: This should be `AZURE_SEARCH_TOP_K` from app config

File: code/tests/functional/tests/backend_api/integrated_vectorization_custom_conversation/test_iv_question_answer_tool.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                "vectorQueries": [
+                    {
+                        "kind": "text",
+                        "k": 5,
```
Comment (cecheta) [inline code/tests/functional/tests/backend_api/integrated_vectorization_custom_conversation/test_iv_question_answer_tool.py:179]: Also here

**Comment by komalg1** [comment]
> I think some of the tests will need updating after now that #939 is in

Updated

**Review by cecheta — APPROVED** [review]
Nice

**Pull request merged**